src/clients/client.ts

```diff
@@ -216,10 +216,18 @@ class Client {
    * @param config - Additional request configuration options
    * @returns Promise containing the search responses
    */
-  async multiSearch<T extends Record<string, any> = Record<string, any>>(
-    queries?: MultiSearchParams,
+  multiSearch<T extends Record<string, unknown> = Record<string, any>>(
+    queries: MultiSearchParams,
+    config?: Partial<Request>,
+  ): Promise<MultiSearchResponse<T>>;
+  multiSearch<T extends Record<string, unknown> = Record<string, any>>(
+    queries: FederatedMultiSearchParams,
+    config?: Partial<Request>,
+  ): Promise<SearchResponse<T>>;
+  async multiSearch<T extends Record<string, unknown> = Record<string, any>>(
+    queries: MultiSearchParams | FederatedMultiSearchParams,
     config?: Partial<Request>,
-  ): Promise<MultiSearchResponse<T>> {
+  ): Promise<MultiSearchResponse<T> | SearchResponse<T>> {
```

Comment:
This seems really overkill to me (and hardly maintainable). I think we can just keep it as it was, and find a way to type `MultiSearchResponse` differently in the `types.ts` file :)

Reply:
If you look into `http-request.ts` we have a ton of overloads there too, I am planning on addressing these eventually.

I have tried typing `queries` with generics so that `multiSearch` would be smaller and simpler, but the problem with that is that we have an optional generic already `T extends Record<string, unknown> = Record<string, any>`, and when you have an optional generic you cannot have type inference anymore, thus typing it with generics won't work.